docs/DeveloperGuide.md

```diff
@@ -257,7 +257,84 @@ The following activity diagram summarizes what happens when a user executes a ne
 * **Alternative 3:** Aliases can map to commands, including other aliases.
     * Pros: User has even more freedom.
     * Cons: Implementation is much more challenging (e.g. how to prevent recursion?).
+    
+
+### Log feature
+
+#### Overview
+
+The number of hours that a user spends at a StudySpot is shown on the GUI. It is saved as a field of `StudySpot`, known
+as `StudiedHours`. This number can be changed by making use of the Log feature.
+
+
+#### Implementation of Log feature
+
+The Log feature is facilitated by `LogCommand.java`, `StudiedHours`.
+
+`StudiedHours.java` is a statistical field in `StudySpot`. It is responsible for storing the hours a user studies at a
+StudySpot. It implements the following operations: 
+- `StudiedHours#isValidLoggedHours()` -- checks if a given input in a Log Command is a positive integer, and that there
+is no integer overflow.
+- `StudiedHours#addHours(StudiedHours toBeAdded)` -- Takes in another `StudiedHours` and adds the two values
+ together, returning a new `StudiedHour`. It also ensures that there is no integer overflow.
+
+`LogCommand.java` is responsible for changing the `StudiedHours` field in a given `StudySpot` (given by its `Name`). 
+It has 3 different ways of doing so:
+- By default, it appends the given input value to the current value.
+    - E.g. If the current StudiedHours is 4, and I log 3 more hours, it adds 4 and 3 together to give 7 hours.
```

Comment:
The uses of "I" here should be changed to "user", to stay consistent with the voice throughout the DG.